addons/google_calendar/models/google_calendar.py

```diff
@@ -350,118 +348,112 @@ def get_one_event_synchro(self, cr, uid, google_id, context=None):
-
-    def update_from_google(self, cr, uid, event, single_event_dict, type, context):
-        if context is None:
-            context = []
-
-        calendar_event = self.pool['calendar.event']
-        res_partner_obj = self.pool['res.partner']
-        calendar_attendee_obj = self.pool['calendar.attendee']
-        calendar_alarm_obj = self.pool['calendar.alarm']
-        user_obj = self.pool['res.users']
-        myPartnerID = user_obj.browse(cr, uid, uid, context).partner_id.id
+        return GsPool._do_request(url, data_json, headers, type='PUT')
+
+    def update_from_google(self, event, single_event_dict, type):
+        CalendarEvent = self.env['calendar.event'].with_context(no_mail_to_attendees=True)
+        ResPartner = self.env['res.partner']
+        CalendarAttendee = self.env['calendar.attendee']
+        CalendarAlarm = self.env['calendar.alarm']
+        my_partner_id = self.env.user.partner_id.id
         attendee_record = []
         alarm_record = set()
-        partner_record = [(4, myPartnerID)]
+        partner_record = [(4, my_partner_id)]
         result = {}
 
-        if self.get_need_synchro_attendee(cr, uid, context=context):
+        if self.get_need_synchro_attendee():
             for google_attendee in single_event_dict.get('attendees', []):
                 partner_email = google_attendee.get('email', False)
                 if type == "write":
                     for oe_attendee in event['attendee_ids']:
-                        if oe_attendee.email == partner_email:
-                            calendar_attendee_obj.write(cr, uid, [oe_attendee.id], {'state': google_attendee['responseStatus'], 'google_internal_event_id': single_event_dict.get('id')}, context=context)
+                        if oe_attendee.email == google_attendee['email']:
+                            CalendarAttendee.browse(oe_attendee.id).write({'state': google_attendee['responseStatus'], 'google_internal_event_id': single_event_dict.get('id')})
                             google_attendee['found'] = True
                             continue
 
                 if google_attendee.get('found'):
                     continue
 
-                attendee_id = res_partner_obj.search(cr, uid, [('email', '=', partner_email)], context=context)
-                if not attendee_id:
+                attendee = ResPartner.search([('email', '=', google_attendee['email'])], limit=1)
+                if not attendee:
                     data = {
                         'email': partner_email,
                         'customer': False,
                         'name': google_attendee.get("displayName", False) or partner_email
                     }
-                    attendee_id = [res_partner_obj.create(cr, uid, data, context=context)]
-                attendee = res_partner_obj.read(cr, uid, attendee_id[0], ['email'], context=context)
+                    attendee = ResPartner.create(data)
+                attendee = attendee.read(['email'])[0]
```

Comment:
Well then we need to create dict here and assign email in it, note that it is used in following lines